Storage/container/one_to_many.h

```diff
@@ -0,0 +1,261 @@
+      InnerAccessor<INNER_MAP> operator*() const { return InnerAccessor<INNER_MAP>(iterator->second); }
+    };
+
+    explicit OuterAccessor(const OUTER_MAP& map) : map_(map) {}
+
+    bool Empty() const { return map_.empty(); }
+    size_t Size() const { return map_.size(); }
+
+    bool Has(const OUTER_KEY& x) const { return map_.find(x) != map_.end(); }
+
+    ImmutableOptional<InnerAccessor<INNER_MAP>> operator[](OUTER_KEY key) const {
+      const auto iterator = map_.find(key);
+      if (iterator != map_.end()) {
+        return std::move(std::make_unique<InnerAccessor<INNER_MAP>>(iterator->second));
+      } else {
+        return nullptr;
+      }
+    }
+
+    OuterIterator begin() const { return OuterIterator(map_.cbegin()); }
+    OuterIterator end() const { return OuterIterator(map_.cend()); }
+  };
+
+  const OuterAccessor<T_FORWARD_MAP> Rows() const { return OuterAccessor<T_FORWARD_MAP>(forward_); }
+
+  const InnerAccessor<T_TRANSPOSED_MAP> Cols() const { return InnerAccessor<T_TRANSPOSED_MAP>(transposed_); }
+
+  Iterator<T_ELEMENTS_MAP> begin() const { return Iterator<T_ELEMENTS_MAP>(map_.begin()); }
+  Iterator<T_ELEMENTS_MAP> end() const { return Iterator<T_ELEMENTS_MAP>(map_.end()); }
+
+ private:
+  void DoErase(const T_KEY& key) {
+    auto& map_row = forward_[key.first];
+    map_row.erase(key.second);
+    if (map_row.empty()) {
+      forward_.erase(key.first);
+    }
+    transposed_.erase(key.second);
+    map_.erase(key);
+  }
+
+  void DoAdd(const T_KEY& key, const T& object) {
+    auto& placeholder = map_[key];
+    placeholder = std::make_unique<T>(object);
+    forward_[key.first][key.second] = placeholder.get();
+    transposed_[key.second] = placeholder.get();
+  }
+
+  T_ELEMENTS_MAP map_;
+  T_FORWARD_MAP forward_;
+  T_TRANSPOSED_MAP transposed_;
+  MutationJournal& journal_;
+};
+
+template <typename T, typename T_UPDATE_EVENT, typename T_DELETE_EVENT>
+using UnorderedOneToMany = GenericOneToMany<T, T_UPDATE_EVENT, T_DELETE_EVENT, Unordered, Unordered>;
+
+template <typename T, typename T_UPDATE_EVENT, typename T_DELETE_EVENT>
+using OrderedOneToMany = GenericOneToMany<T, T_UPDATE_EVENT, T_DELETE_EVENT, Ordered, Ordered>;
+
+}  // namespace container
+
+template <typename T, typename E1, typename E2>  // Entry, update event, delete event.
+struct StorageFieldTypeSelector<container::UnorderedOneToMany<T, E1, E2>> {
+  static const char* HumanReadableName() { return "UnorderedOneToMany"; }
+};
+
+template <typename T, typename E1, typename E2>  // Entry, update event, delete event.
+struct StorageFieldTypeSelector<container::OrderedOneToMany<T, E1, E2>> {
+  static const char* HumanReadableName() { return "OrderedOneToMany"; }
+};
+
+}  // namespace storage
+}  // namespace current
+
+using current::storage::container::UnorderedOneToMany;
+using current::storage::container::OrderedOneToMany;
+
+#endif  // CURRENT_STORAGE_CONTAINER_ONE_TO_MANY_H
```

Comment:
Looks good.
Note to self: I've reviewed this file by diff-ing it against `one_to_one.h`.